axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/axon/DefaultPersistentStreamMessageSourceFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.axonserver.connector.event.axon;
+
+import io.axoniq.axonserver.connector.event.PersistentStreamProperties;
+import org.axonframework.config.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class DefaultPersistentStreamMessageSourceFactory implements PersistentStreamMessageSourceFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultPersistentStreamMessageSourceFactory.class);
+    private final Set<String> usedNames = new HashSet<>();
```

Comment:
This implementation is not concurrency-safe. Please use a Set implementation from the `java.util.concurrent` package.

Reply:
At first, I assumed we didn't need thread-safety because it's only invoked during configuration. However, you're right - since this class is public, we can't know when or how someone might use or inject it.